dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/WorkerClusters.java

```diff
@@ -134,6 +145,17 @@ public void onServerAdded(WorkerServerMetadata workerServer) {
     @Override
     public void onServerRemove(WorkerServerMetadata workerServer) {
         workerMapping.remove(workerServer.getAddress(), workerServer);
+        synchronized (workerGroupMapping) {
+            List<String> removeWorkerGroupAddrList = workerGroupMapping.get(workerServer.getWorkerGroup());
+            if (removeWorkerGroupAddrList != null && removeWorkerGroupAddrList.contains(workerServer.getAddress())) {
+                removeWorkerGroupAddrList.remove(workerServer.getAddress());
+                if (removeWorkerGroupAddrList.isEmpty()) {
+                    workerGroupMapping.remove(workerServer.getWorkerGroup());
+                } else {
+                    workerGroupMapping.put(workerServer.getWorkerGroup(), removeWorkerGroupAddrList);
+                }
```

Comment:
```suggestion
                }
```
No need to put again.

Reply:
fixed.